packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain/DestinationNetworkBox.tsx

```diff
@@ -29,25 +29,103 @@ import {
   NetworkSelectionContainer
 } from '../../common/NetworkSelectionContainer'
 
-export function DestinationNetworkBox({
+function DestinationChainBalance({
   customFeeTokenBalances,
   showUsdcSpecificInfo
 }: {
   customFeeTokenBalances: Balances
   showUsdcSpecificInfo: boolean
 }) {
-  const { address: walletAddress } = useAccount()
+  const {
+    app: { selectedToken }
+  } = useAppState()
   const [networks] = useNetworks()
   const { childChain, childChainProvider, isDepositMode } =
     useNetworksRelationship(networks)
   const { isArbitrumOne } = isNetwork(childChain.id)
-  const {
-    app: { selectedToken }
-  } = useAppState()
+
+  const selectedTokenBalances = useSelectedTokenBalances()
   const { ethParentBalance, ethChildBalance, erc20ChildBalances } =
     useBalances()
-  const selectedTokenBalances = useSelectedTokenBalances()
+
   const nativeCurrency = useNativeCurrency({ provider: childChainProvider })
+
+  if (selectedToken) {
```

Comment:
We are only showing balance for whatever token is selected, so it was cleaner to move it to it's own method to avoid having complex ternary chains